### PR TITLE
Fix paths on windows

### PR DESCRIPTION
**Pull request by daniilprohorov**

### CODE REVIEW

**Comment by ryukzak** [comment]
Hi, sorry for the long review!

In `NITTA.Utils` you can find: `toModuleName`. This function should transform the text to the correct module name. Can you improve this function and add it to the code where needed?

Also, can you revoke the `testcaseM` function, because I see no reason why we shouldn't use `toModuleName` everywhere?

File: test/NITTA/Utils/Tests.hs [hunk 1]
```diff
@@ -37,3 +38,8 @@ tests =
             source ["b", "a"] == source ["a", "b"] @? "Source eq"
             source ["b", "a"] /= source ["a", "c"] @? "Source not eq"
         ]
+
+modifyTestName name = replace " " "_" $ replace ":" "_" name
```
Comment (ryukzak) [inline test/NITTA/Utils/Tests.hs:42]: Rename `modifyTestName` to `fixModuleName`

**Review by ryukzak — CHANGES_REQUESTED** [review]
- Please, fix code style.
- I don't think that we need to have `testCaseM` function. It should be a part of base `testCase`. Also, looks like you can reduce amount of usage "fix funciton" in the code.

**Comment by daniilprohorov** [comment]
image
Are you sure, that we have to use toModuleName everywhere?

File: src/NITTA/Project.hs [hunk 2]
```diff
@@ -162,10 +162,12 @@ writeImplementation prj@Project{pTargetProjectPath = prjPath, pInProjectNittaPat
...
     where
         copyLibraryFile Project{pTargetProjectPath, pInProjectNittaPath, pLibPath} file = do
-            let fullNittaPath = pTargetProjectPath </> pInProjectNittaPath
-            source <- makeAbsolute $ joinPath [pLibPath, file]
-            target <- makeAbsolute $ joinPath [fullNittaPath, "lib", file]
-            createDirectoryIfMissing True $ takeDirectory target
+            let fullNittaPath = joinPath [pTargetProjectPath, pInProjectNittaPath]
+            source <- makeAbsolute $ normalise $ joinPath [pLibPath, file]
+            target <- makeAbsolute $ normalise $ joinPath [fullNittaPath, "lib", file]
+            directory <- makeAbsolute $ normalise $ joinPath [fullNittaPath, "lib", takeDirectory file]
```
Comment (ryukzak) [inline src/NITTA/Project.hs:168]: Also, can you import `normalise` explicitly or with qualified name?

Please, add small issue to grep our project and find other places when it is reasonable to `normalise` paths for new project members.

File: test/NITTA/Frontends/Lua/Tests.hs [hunk 3]
```diff
@@ -653,7 +653,7 @@ test_examples =
             pid(0, 0)
             |]
     , -- FIXME: uncomment when IO synchronization propogation and SPI will be fixed.
-      -- , testCase "examples/fibonacci.lua drop" $ either assertFailure return
+      -- , testCaseM "examples/fibonacci.lua drop" $ either assertFailure return
```
Comment (ryukzak) [inline test/NITTA/Frontends/Lua/Tests.hs:656]: You miss this in the comment:

> I don't think that we need to have testCaseM function. It should be a part of base testCase. Also, looks like you can reduce amount of usage "fix funciton" in the code.

**Review by ryukzak — CHANGES_REQUESTED** [review]

File: test/NITTA/Model/ProcessorUnits/Tests/DSL.hs [hunk 4]
```diff
@@ -187,8 +187,8 @@ unitTestCase ::
     u ->
     StateT (UnitTestState u v x) IO () ->
     TestTree
-unitTestCase name pu alg = testCase name $ do
-    void $ evalUnitTestState name pu alg
+unitTestCase name pu alg = testCase (toModuleName name) $ do
```
Comment (ryukzak) [inline test/NITTA/Model/ProcessorUnits/Tests/DSL.hs:190]: We don't need it here.

**Review by ryukzak — APPROVED** [review]

**Pull request merged**